gdnative-core/src/export/method.rs

```diff
@@ -284,16 +287,303 @@ impl<'a> Varargs<'a> {
+    ///     args.check_length(2..=2)?;
+    ///     let a: usize = args.get(0)?;
+    ///     let b: i64 = args.get(1)?;
+    /// #   Ok(())
+    /// # }
+    /// ```
+    #[inline]
+    pub fn get<T: FromVariant>(&self, index: usize) -> Result<T, ArgumentTypeError> {
+        let relative_index = index;
+        let actual_index = index + self.offset_index;
+
+        match self.args.get(relative_index) {
+            Some(v) => match T::from_variant(v) {
+                Ok(ok) => Ok(ok),
+                Err(err) => Err(ArgumentTypeError::new(actual_index, err)),
+            },
+            None => {
+                let err = FromVariantError::Custom("Argument is not set".to_owned());
+                Err(ArgumentTypeError::new(actual_index, err))
+            }
+        }
+    }
+
+    /// Returns the type-converted value at the specified argument position.
+    /// Returns `None` if the argument is not set.
+    ///
+    /// # Errors
+    /// Returns an error if the conversion fails.
+    ///
+    /// # Examples
+    /// ```
+    /// # fn foo(args: gdnative::export::Varargs) -> Result<(), Box<dyn std::error::Error>> {
+    ///     args.check_length(1..=2)?;
+    ///     let a: usize = args.get(0)?;
+    ///     let b: i64 = args.get_opt(1)?.unwrap_or(72);
+    /// #   Ok(())
+    /// # }
+    /// ```
+    #[inline]
+    pub fn get_opt<T: FromVariant>(&self, index: usize) -> Result<Option<T>, ArgumentTypeError> {
+        let relative_index = index;
+        let actual_index = index + self.offset_index;
+
+        match self.args.get(relative_index) {
+            Some(v) => match T::from_variant(v) {
+                Ok(ok) => Ok(Some(ok)),
+                Err(err) => Err(ArgumentTypeError::new(actual_index, err)),
+            },
+            None => Ok(None),
         }
     }
+
+    /// Returns the type-converted value from the specified argument position.
+    /// Can be converted to any type that implements TryFrom<Varargs>.
+    ///
+    /// # Errors
+    /// Returns an error if the conversion fails.
+    ///
+    /// # Examples
+    /// ```ignore
+    /// # fn foo(args: gdnative::export::Varargs) -> Result<(), Box<dyn std::error::Error>> {
+    ///     args.check_length(1..)?;
+    ///     let a: usize = args.get(0)?;
+    ///     let rest: Vec<i64> = args.get_rest(1)?;
+    /// #   Ok(())
+    /// # }
+    /// ```
+    #[inline]
+    pub fn get_rest<T>(&self, rest_index: usize) -> Result<T, <Varargs<'a> as TryInto<T>>::Error>
+    where
+        Varargs<'a>: TryInto<T>,
+    {
+        let relative_rest_index = rest_index;
+        let actual_rest_index = rest_index + self.offset_index;
+
+        let rest = self.args.get(relative_rest_index..).unwrap_or_default();
+        let varargs = Varargs::<'a> {
+            idx: 0,
+            args: rest,
+            offset_index: actual_rest_index,
+        };
+        varargs.try_into()
+    }
+
+    /// Get the varargs's offset index.
+    #[inline]
+    #[must_use]
```

Comment:
I would reserve `#[must_use]` for situations, where both:
1. there is a high chance that the user accidentally forgets a return value
2. such a forgetting can cause bugs, e.g. by swallowing errors

I don't think any of the uses in this file qualifies for these criteria. Simple getters or even `new()` do generally not fall in that category, otherwise we'd have to annotate half the library with `#[must_use]`.

Some examples where we used it:
* `fn test_something() -> bool` in Godot integration tests
   *  forgetting to check the result would allow tests to fail silently.
* [`MethodBuilder`](https://godot-rust.github.io/docs/gdnative/export/struct.MethodBuilder.html) (struct attribute)
   *  a user could register a method and miss the final `done()` invocation.

Reply:
For example, code like this.
```rust
args.offset_index(); // No operation
```
This would be unintended behavior for the person who wrote this, so it makes sense to warn them.

> Simple getters or even new() do generally not fall in that category, otherwise we'd have to annotate half the library with #[must_use].

The standard library recently did that. So even `godot-rust` will have to do so by the time the version reaches `1.0.0`.

Actually, when I auto-generated the getter, it just generated the `#[must_use]` as well, which I didn't intend. :upside_down_face: